desc/compute/_profiles.py

```diff
@@ -638,59 +638,161 @@ def _iota_rr(params, transforms, profiles, data, **kwargs):
+    current_r = profiles["current"].compute(transforms["grid"], params["c_l"], dr=1)
+    data["iota_num current"] = (
         2
         * jnp.pi
         * mu_0
-        * profiles["current"].compute(transforms["grid"], params["c_l"], dr=0)
-        / data["psi_r"]
+        * transforms["grid"].replace_at_axis(
+            current / data["psi_r"],
+            lambda: current_r / data["psi_rr"],
+        )
     )
-    beta = surface_integrals(
-        transforms["grid"],
+    return data
+
+
+@register_compute_fun(
+    name="iota_num vacuum",
+    label="\\iota_{\\mathrm{numerator}}~\\mathrm{in~vacuum}",
+    units="m^{-1}",
+    units_long="inverse meters",
+    description="Numerator of rotational transform formula, vacuum contribution",
+    dim=1,
+    params=[],
+    transforms={"grid": []},
+    profiles=[],
+    coordinates="r",
+    data=["lambda_z", "g_tt", "lambda_t", "g_tz", "sqrt(g)"],
+    axis_limit_data=["g_tz_r", "sqrt(g)_r"],
+)
+def _iota_num_vacuum(params, transforms, profiles, data, **kwargs):
+    """Vacuum contribution to the numerator of rotational transform formula."""
+    iota_num_vacuum = transforms["grid"].replace_at_axis(
         (data["lambda_z"] * data["g_tt"] - (1 + data["lambda_t"]) * data["g_tz"])
         / data["sqrt(g)"],
+        lambda: -(1 + data["lambda_t"]) * data["g_tz_r"] / data["sqrt(g)_r"],
     )
-    data["iota_num"] = transforms["grid"].replace_at_axis(alpha + beta, 0)
+    data["iota_num vacuum"] = surface_integrals(transforms["grid"], iota_num_vacuum)
     return data
 
 
 @register_compute_fun(
-    name="iota_num_r",
-    label="\\partial_{\\rho} \\iota_{\\mathrm{numerator}}",
+    name="iota_num_r current",
+    label="\\partial_{\\rho} \\iota_{\\mathrm{numerator}}~\\mathrm{from~current}",
     units="m^{-1}",
     units_long="inverse meters",
-    description="Numerator of rotational transform formula, first radial derivative",
+    description="Numerator of rotational transform formula, current contribution, "
+    + "first radial derivative",
     dim=1,
     params=["c_l"],
     transforms={"grid": []},
     profiles=["current"],
     coordinates="r",
+    data=["0", "psi_r", "psi_rr"],
+)
+def _iota_num_r_current(params, transforms, profiles, data, **kwargs):
+    if profiles["current"] is None:
+        data["iota_num_r current"] = jnp.nan * data["0"]
+        return data
+
+    # 4π^2 I = 4π^2 (mu_0 current / 2π) = 2π mu_0 current
+    current = profiles["current"].compute(transforms["grid"], params["c_l"], dr=0)
+    current_r = profiles["current"].compute(transforms["grid"], params["c_l"], dr=1)
+    current_rr = profiles["current"].compute(transforms["grid"], params["c_l"], dr=2)
+    data["iota_num_r current"] = (
+        2
+        * jnp.pi
+        * mu_0
+        * transforms["grid"].replace_at_axis(
+            (current_r * data["psi_r"] - current * data["psi_rr"]) / data["psi_r"] ** 2,
+            lambda: current_rr / (2 * data["psi_rr"]),
```

Comment:
just linking to document change https://github.com/PlasmaControl/DESC/pull/642#discussion_r1310570548